bin/run_docker.py

```diff
@@ -117,26 +118,42 @@ def create_log_file(
     If no text is given, it writes "No Logs" to the file.
 
     Arguments:
-        log_filename: The name of the log file to create
-        log_text: The text to write to the log file. If given as a byte string,
-                  it will be decoded as UTF-8 before being written.
+        log_file_name: The name of the log file to create
+        log_max_size: The maximum size of the log file in kilobytes
+        log_file_path: The path where the log file will be created.
+                       If not specified, the current working directory will be used.
+        log_text: The text to write to the log file
 
     """
-    if not log_file_path:
+    if log_file_path is None:
         log_file_path = os.getcwd()
 
+    if log_text is None:
+        log_text = "No Logs"
+    elif isinstance(log_text, bytes):
+        log_text = log_text.decode("utf-8")
+
+    # We need the size of the log message in bytes
+    log_text_size = len(log_text.encode("utf-8"))
+
+    # Truncate the log message if it exceeds the maximum size
+    if log_text_size > log_max_size * 1000:
+        print(f"Original log message exceeds {log_max_size} Kb. Truncating...")
+        log_text = log_text[-(log_max_size * 1000) :]
```

Comment:
Will want to confirm the behavior of this for the log lines that are at the edge of the cut-off. I imagine this will cut the line into a partial log.

Reply:
Currently, yes, a word that's at the edge of the cutoff will be sliced halfway and leave a partial word. Is this what you meant?

<img width="514" alt="image" src="https://github.com/Sage-Bionetworks-Workflows/nf-synapse-challenge/assets/32107699/b2d52e51-5395-48e2-aae8-48ffe2f792fd">

My first crack at a solution is to modify the `log_max_size` arg to avoid slicing words:

<img width="319" alt="image" src="https://github.com/Sage-Bionetworks-Workflows/nf-synapse-challenge/assets/32107699/b50a88cb-74a3-413b-b817-b7c699a844a1">

There are 2 assumptions I'm making with this logic:
1. Index is a negative number, which should always be the case since it's just the len of a string or bytes
2. Spaces are used around complete words only (no spaces within a word)

Reply:
> Currently, yes, a word that's at the edge of the cutoff will be partly sliced and leave a partial word. Is this what you meant?

Yup! Thanks for confirming what the behavior of this is. I think we have a few options on how we want to proceed here:

1. Leave as is
2. Truncate with floor of ceiling of the line, or the word

I am leaning towards leaving as it; Only if there a problem spend more time to revisit this.

Reply:
We'll be updating the Challenge participant documentation letting them know that they may get back partial logs if it exceeds 50KB, but I see the value in preventing the random word cutoffs, so I made a [Jira ticket](https://sagebionetworks.jira.com/browse/ORCA-319) for this as an improvement to the workflow for after PEGS. Thanks for pointing this out!